aws/cloudformation/iam.yml.erb

```diff
@@ -390,6 +392,32 @@ roles.each do |name, config| %>
           Value: codedotorg
 <%- end -%>
 
+  CloudformationPassRolePolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      ManagedPolicyName: CloudformationPassRolePolicy
+      Path: /admin/
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action: iam:PassRole
+            Resource: '*'
+            Condition:
+              StringEquals:
+                iam:PassedToService: cloudformation.amazonaws.com
+          # Require CloudFormation Service Role on all stack operations.
```

Comment:
What is the effect of this deny rule? Ultimately to me it looks like we're denying create, update, and delete and create change set for everything except a certain transform.

I'm not quite sure what the string not equals is doing. Overall it just doesn't seem to add up to what the description says of requiring it on all stack operations.  Or my brain is twisted up like it does, with negative-rule logic. Can we update the comment to maybe explain this better?

Reply:
It denies cloudformation stack modification unless you tell Cloudformation to assume a certain role for it's stack modifications. Normally you'll just run these operations with your own role. I think the CloudFormationService role is _more_ powerful than the engineer's role, and it gives us a place to grant permissions useful for CI/CD that we don't necessarily want to grant to everyone.

- but it lets you work with serverless transforms (not 100% sure the scenario here, but surely related to Lambda custom resources, or perhaps Javabuidler)

It's an IAM setup that predates me, so I can't fully speak to the original intent.